kythe/go/localrun/cmd/main.go

```diff
@@ -0,0 +1,195 @@
+
+"bazel build" and generate cross references for just go and java code in the whole repo:
+
+	localrun --language=go,java //...
+
+Since this uses a standard bazel build invocation to build the targets, any targets that are
+included in the path will be included. If you would like to limit the build to just java_library
+targets you can employ bazel query to this end.
+
+Build only java_library targets and cross reference only java code:
+
+bazel query 'kind(java_library, //...)' | xargs localrun
+`)
+}
+
+// main entrypoint for the program.
+func main() {
+	wd, _ := os.Getwd()
+	cacheDir, _ := os.UserCacheDir()
+
+	flag.Var(&languages, "language", "Case insensitive list of languages that should be extracted and indexed")
+	flag.IntVar(&port, "port", 8080, "The port to start the http server on once everything is finished")
+	flag.StringVar(&workingDir, "working_dir", wd, "The directory for all bazel oprations to begin relative to")
+	flag.StringVar(&kytheRelease, "kythe_release", "/opt/kythe", "The directory that holds a Kythe release. Releases can be downloaded from https://github.com/kythe/kythe/releases")
+	flag.StringVar(&outputDir, "output_dir", filepath.Join(cacheDir, "output"), "The directory to create intermediate artifacts in")
+	flag.DurationVar(&indexingTimeout, "indexing_timeout", 300*time.Second, "How long to wait before indexing a compilation unit times out")
+	flag.Parse()
+
+	targets, err := getTargets(flag.Args())
+	if err != nil {
+		usage()
+		log.Fatalf("Error invoking localrun: %v\n", err)
+	}
+
+	log.Printf("Building %v for targets: %s\n",
+		languages.LanguageSet.String(), strings.Join(targets, " "))
+
+	r := &localrun.Runner{
+		KytheRelease: kytheRelease,
+		WorkingDir:   workingDir,
+		OutputDir:    outputDir,
+		CacheSize:    cacheSize,
+
+		//WorkerPoolSize: runtime.GOMAXPROCS(0) * 2,
+		WorkerPoolSize: runtime.GOMAXPROCS(0)*0 + 1,
+
+		Languages: languages.LanguageSet,
+		Targets:   targets,
+
+		Timeout: 300 * time.Second,
+
+		Port: port,
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	if err = r.Extract(ctx); err != nil {
+		log.Fatalf("Error extracting: %v\n", err)
+	}
+
+	if err = r.Index(ctx); err != nil {
+		log.Fatalf("Error indexing: %v\n", err)
+	}
+
+	if err = r.PostProcess(ctx); err != nil {
+		log.Fatalf("Error post processing: %v\n", err)
+	}
+
+	if err = r.Serve(ctx); err != nil {
+		log.Fatalf("Error starting server: %v\n", err)
+	}
+
+	log.Println("Finished successfully.")
+}
+
+// languageFlag is a flag.Value that accepts either a list of languages or repeated
+// languages for the same flag value.
+type languageFlag struct {
+	localrun.LanguageSet
+}
+
+// String implements flag.Value.
+func (lf *languageFlag) String() string {
+	return lf.LanguageSet.String()
+}
+
+// Set implements flag.Value.
+func (lf *languageFlag) Set(value string) error {
+	for _, v := range strings.Split(value, ",") {
+		l, ok := localrun.LanguageMap[v]
```

Comment:
Add a `local.KnownLanguage(v)` function to abstract this map check.

Reply:
Done